src/main/java/roomescape/business/domain/Reservation.java

```diff
@@ -0,0 +1,84 @@
+package roomescape.business.domain;
+
+import java.time.LocalDate;
+import java.util.Objects;
+
+public class Reservation {
+
+    private Long id;
+
+    private final String name;
+    private final LocalDate date;
+    private final Time time;
+
+    public Reservation(final String name, final LocalDate date, final Time time) {
+        validateNonNull(name, date, time);
+        validateNameIsNotBlank(name);
+        this.name = name;
+        this.date = date;
+        this.time = time;
+    }
+
+    public Reservation(final Long id, final String name, final LocalDate date, final Time time) {
+        validateNonNull(id, name, date, time);
+        validateNameIsNotBlank(name);
+        this.id = id;
+        this.name = name;
+        this.date = date;
+        this.time = time;
+    }
```

Comment:
id는 nullable하다고 생각하는데요. 
차라리 오버로딩한 생성자는 생성자를 체이닝하여 호출하도록 변경하고 
DB를 읽어 id가 non-nullable한 객체를 만들 때는 정적 팩토리 메서드 등을 만들어서 id에 대한 null 유효성 검사를 별도로 수행하도록 하면 어떨까요? 

유사한 유효성 검증 로직이 여기저기 퍼져있어 추후 변경하거나 유지보수하기가 조금 어려워질 것 같다는 생각이들기도해서요. 🤔 

```suggestion
    public Reservation(final String name, final LocalDate date, final Time time) {
        this(null, name, date, time);
    }

    public Reservation(final Long id, final String name, final LocalDate date, final Time time) {
        validateNonNull(id, name, date, time);
        validateNameIsNotBlank(name);
        this.id = id;
        this.name = name;
        this.date = date;
        this.time = time;
    }
```

Reply:
> 유사한 유효성 검증 로직이 여기저기 퍼져있어 추후 변경하거나 유지보수하기가 조금 어려워질 것 같다는 생각이들기도해서요. 🤔

인정합니다. 반영하겠습니다 👍

Reply:
> 유사한 유효성 검증 로직이 여기저기 퍼져있어 추후 변경하거나 유지보수하기가 조금 어려워질 것 같다는 생각이들기도해서요. 🤔

인정합니다.
추가적으로 id를 포함한 생성자가 public일 경우 의도치않은 id null 객체가 생성될 수 있을 것 같아 private으로 막고, id를 포함하여 객체를 생성할 때는 정적 팩토리 메서드만을 이용하게 하였습니다.

77efbdf3b6d355de887fe625102fc0f686a46b47 0e34664e7169bd2ff6a43d0c66c29e20efb8d2c7